connectivity/suite.go

```diff
@@ -21,13 +21,80 @@ import (
 	"github.com/cilium/cilium-cli/connectivity/tests"
 )
 
+var (
+	//l3Policies = check.PolicyContext{}
+)
+
 func Run(ctx context.Context, k *check.K8sConnectivityCheck) error {
 	return k.Run(ctx,
+		// This should fail, failures not yet handled properly:
+		(&tests.PodToPod{Variant: "-client-egress-only-dns-SHOULD-FAIL"}).WithPolicy(clientEgressOnlyDNSPolicyYaml),
+		// Policy installed with 'WithPolicy()' is automatically removed, so this should succeed:
 		&tests.PodToPod{},
+		// This policy allows port 8080 from client to echo, so this should succeed
+		(&tests.PodToPod{Variant: "-client-egress-to-echo"}).WithPolicy(clientEgressToEchoPolicyYaml),
+		// This should also succeed, no policy applied
+		&tests.PodToPod{Variant: "-2"},
+		// Apply policy that is kept around until explicitly removed
+		&check.CiliumNetworkPolicy{Title: "client-egress-only-dns", Policy: clientEgressOnlyDNSPolicyYaml},
+		&tests.PodToPod{Variant: "-client-egress-only-dns-SHOULD-FAIL-too"},
+		&check.CiliumNetworkPolicy{}, // delete all applied policies
+		// This should also succeed, no policy applied
+		&tests.PodToPod{Variant: "-3"},
 		&tests.PodToService{},
 		&tests.PodToNodePort{},
 		&tests.PodToLocalNodePort{},
 		&tests.PodToWorld{},
 		&tests.PodToHost{},
 	)
 }
+
+var (
+	clientEgressOnlyDNSPolicyYaml = `
+apiVersion: cilium.io/v2
+kind: CiliumNetworkPolicy
+metadata:
+  namespace: cilium-test
+  name: client-egress-only-dns
+spec:
+  endpointSelector:
+    matchLabels:
+      kind: client
+  egress:
+  - toPorts:
+    - ports:
+      - port: "53"
+        protocol: ANY
+    toEndpoints:
+    - matchLabels:
+        k8s:io.kubernetes.pod.namespace: kube-system
+        k8s:k8s-app: kube-dns
+`
+	clientEgressToEchoPolicyYaml = `apiVersion: cilium.io/v2
```

Comment:
These could go into `.yaml` files so we can get syntax highlighting and proper tab support. The new `//go:embed` directive can be used so the yamls don't need to be shipped with the test binary and no extra code is required to read them from disk.

Reply:
Thanks for the pointer, YAMLs as Go strings are no fun to edit :-)